apps/wallet/src/background/keyring/Account.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { Keypair, SuiAddress } from '@mysten/sui.js';
+
+export type AccountType = 'derived' | 'imported';
+
+export class Account {
+    #keypair: Keypair;
+    public readonly derivationPath: string | null;
+    public readonly address: SuiAddress;
+
+    constructor(
+        type: 'derived',
```

Comment:
I'd probably just make a union type for an account instead of having function overloads. E.g.:
```typescript
type Account = {
  type: 'derived';
  derivationPath: string;
  keypair: Keypair;
} | {
  type: 'imported';
  keypair: Keypair;
}
```

Honestly if you do that, you might not even need this `Account` class at all, you could just have utility functions that operate over that union instead.

Reply:
Removed the overloads, I will keep the class to make easier adding methods and extra data that is used in the other PRs